src/client/Components/Pages/Dashboard/ActionButton.jsx

```diff
@@ -25,11 +25,33 @@ export default class ActionButton extends React.Component<ActionButtonProps, Act
     (this: any).changeButtonState = this.changeButtonState.bind(this);
   }
 
-  changeButtonState() {
+  async changeButtonState() {
     if (this.state.status === 'active') {
-      this.setState({ status: 'paused', isActive: false });
+      await new Promise((resolve) => this.setState({ status: 'paused', isActive: false }, resolve));
```

Comment:
nit: there may be cleaner way to do this

Reply:
I've removed state from the action button, no longer relevant